packages/slate-editor/src/extensions/tables/onKeyDown.ts

```diff
@@ -0,0 +1,52 @@
+import { TablesEditor } from '@prezly/slate-tables';
+import { isHotkey } from 'is-hotkey';
+import type { KeyboardEvent } from 'react';
+import { type Element, Editor, Range, Transforms } from 'slate';
+
+const isClipboardCopy = isHotkey(['mod+c', 'ctrl+insert']);
+const isClipboardCut = isHotkey(['mod+x', 'shift+delete']);
+
+export function onClipboardHotkey(
+    event: KeyboardEvent,
+    editor: Editor & TablesEditor,
+    createDefaultElement: () => Element,
+) {
+    const selection = editor.selection;
+
+    if (!selection || Range.isExpanded(selection)) return;
+
+    if (isClipboardCopy(event)) {
+        const entry = TablesEditor.findParentTable(editor, selection);
+        if (!entry) {
+            return;
+        }
+        const [, path] = entry;
+
+        Transforms.select(editor, path); // Select the table
+
+        document.execCommand('copy');
+    }
+
+    if (isClipboardCut(event)) {
+        const entry = TablesEditor.findParentTable(editor, selection);
+        if (!entry) {
+            return;
+        }
+        const [, path] = entry;
+
+        Editor.withoutNormalizing(editor, () => {
+            Transforms.select(editor, path); // Select the table
+            const ref = Editor.pathRef(editor, path);
+
+            if (document.execCommand('cut')) {
```

Comment:
if you select several cells and then cut them the cells will be removed as well and table can be shifted. I think we need just copy selected content and then remove selected nodes, without cells

Reply:
> Why do we need to copy whole table if we select some cells?

Hm... For me, it never does what I expect (in Firefox).
However, I've tried it in Chrome and it's much better. So probably you're right that it's better to revert it.

> I think we need to just copy selected content and then remove selected nodes, without cells

Oh, good idea. I like it :+1: Will do.

Reply:
> if you select several cells and then cut them the cells will be removed as well and table can be shifted

This particular code here is only running when the selection is collapsed. 

- If the cursor is inside a cell, the selection is collapsed, and "copy" hotkey is pressed, we copy the whole table
- If the cursor is inside a cell, the selection is collapsed, and "cut" hotkey is pressed, we cut the whole table

I believe this consistency does make sense.

> Copy the full table if multiple cells are selected

This part was implemented in `withTablesCopyPasteBehaviour` and is reverted by eb1bfd91c0d4b1011f7eea582ccb7c217a81aafe.

Thanks!